packages/web-app/src/middleware/check-appeal-outcome.spec.js

```diff
@@ -0,0 +1,39 @@
+const { mockReq, mockRes } = require('../../test/utils/controller-mocks');
+const checkAppealOutcome = require('./check-appeal-outcome');
+
+describe('middleware/check-appeal-outcome', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+  [
+    {
+      title: 'Return 404 if there is no appeal id in the params',
+      req: { ...mockReq, params: {} },
+      expected: (req, res, next) => {
+        expect(next).not.toHaveBeenCalled();
+        expect(res.sendStatus).toHaveBeenCalledWith(404);
+      },
+    },
+    {
+      title: 'Return next if there is appeal id in the params',
+      req: {
+        ...mockReq,
+        session: {},
+        params: {
+          appealId: 12345,
+        },
+      },
+      expected: (req, res, next) => {
+        expect(next).toBeCalledTimes(1);
```

Comment:
We could include an assertion here to check that `req.session.outcome = 'INCOMPLETE'`

Reply:
The reason for the assertion is that in https://github.com/Planning-Inspectorate/back-office/pull/30/files#diff-e01496a939d3d50a4f28156c3afc14aac16ccb48a9f1614e0cb9081d0951d3d0R16 we have `req.session.outcome = 'INCOMPLETE';` which is going to happen when this function is called assuming there is an appeal id so the assertion is to check that this value is set when `next()` is called.

Reply:
I understand that, the reason why this was set to incomplete was just me hardcoding the outcome, I would imagine that value would come from the db when we refactor this, so it wasn't done with any specific logic in place.